internal/rules/affects.go

```diff
@@ -45,3 +48,56 @@ func CheckAffectedProduct(json *string) []ValidationError {
 	}
 	return errors
 }
+
+// Invalid version string examples:
+// - "n/a" or "unspecified"
+// - anything that includes whitespace, e.g. "v12.07 and earlier"
+// - special characters like "<" or ","
+var validVersionRe = regexp.MustCompile(`^(\*|[a-zA-Z0-9]+[-_:\.a-zA-Z0-9]*)$`)
+
+func CheckInvalidVersion(json *string) []ValidationError {
```

Comment:
Let's add some [doc comments](https://go.dev/doc/comment) for more awesome!

```suggestion
CheckInvalidVersion returns an array of detected version-related ValidationError findings.
It checks that the affected.versions sub-fields are used correctly.
func CheckInvalidVersion(json *string) []ValidationError {
```

Reply:
Added!